### PR TITLE
NavBar Component & Search Modal (Desktop, until 795px)

### LINKED ISSUE #2: Navbar Component
Complete the navbar component. This should include routing between pages, the dropdown for when the account button is clicked, and the designs for the searchbar (doesn't have to be functional).

There is already an existing navbar component in the component directory, so feel free to use this as a starting point (it is very barebones).

### PR DESCRIPTION
**Pull request by aksharans**
### Administrative Info
[GitHub Issue #2 ] [GitHub Issue #21 ]
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- created navbar component which contains the logo, page links, a search component, and profile icon 
- search bar opens search modal, where you can search for files on test data
- profile icon opens up a dropdown modal

### Testing
How did you confirm your changes worked? 
- tested using/displaying the components and resizing until 795px

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

https://user-images.githubusercontent.com/51723201/150698619-973cd511-7516-4abf-8f55-5ae12e4a74d2.mov

### CODE REVIEW

File: frontend/src/components/Search.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    id: "file_2",
+    name: "TSE File",
+  },
+  {
+    id: "file_3",
+    name: "Testing",
+  },
+];
+
+function FileResult({ val, index }) {
```
Comment (PatrickBrown1) [inline frontend/src/components/Search.js:21]: This looks pretty similar to the other file component, should we abstract it into a separate component and use it in the modal and on the file pages?

File: frontend/src/constants/links.js [hunk 2]
```diff
@@ -6,4 +6,5 @@ export class SITE_PAGES {
   static WISH_GRANTING = "/wish-granting";
   static PROFILE = "/profile";
   static PEOPLE = "/people";
+  static MANAGE = "/manage";
```
Comment (PatrickBrown1) [inline frontend/src/constants/links.js:9]: I meant for the "people" page to be the manage page. Manage probably makes more sense, but we can get rid of the people components at this point if you've replaced them

**Review by PatrickBrown1 — CHANGES_REQUESTED** [review]
^^ No major changes, just some little admin and style stuff. I'll let Andrew be another set of eyes

File: frontend/src/components/NavBar.js [hunk 3]
```diff
@@ -1,17 +1,92 @@
...
-      <a href={SITE_PAGES.LOGIN}>Log In</a>
-      <a href={SITE_PAGES.PEOPLE}>People</a>
-      <a href={SITE_PAGES.WISH_GRANTING}>Wish Granting</a>
-      <a href={SITE_PAGES.PROFILE}>Profile</a>
-    </div>
+    <nav className="container">
+      <ul className="navlist">
+        {/* Container for Logo and page links */}
+        <li className="logo-and-pages">
+          <li className="logo-container">
```
Comment (Cubified) [inline frontend/src/components/NavBar.js:28]: React complains if you nest `<li>` tags inside of each other, changing these to divs fixes the error and keeps the accessibility of using `<nav>`/`<ul>`/`<li>`/etc.

```suggestion
          <div className="logo-container">
```

File: frontend/src/components/NavBar.js [hunk 4]
```diff
@@ -1,17 +1,92 @@
...
+            <NavLink to={SITE_PAGES.MANAGE}>
+              <img src="/img/maw_logo.png" alt="Make-A-Wish logo" className="maw-logo" />
+            </NavLink>
+          </li>
```
Comment (Cubified) [inline frontend/src/components/NavBar.js:32]:
```suggestion
          </div>
```

File: frontend/src/components/NavBar.js [hunk 5]
```diff
@@ -1,17 +1,92 @@
...
+
+          <li className="pages-container">
```
Comment (Cubified) [inline frontend/src/components/NavBar.js:34]:
```suggestion
          <div className="pages-container">
```

File: frontend/src/components/NavBar.js [hunk 6]
```diff
@@ -1,17 +1,92 @@
...
+            {Object.entries(PAGES).map(([page, route]) => (
+              <NavLink className="page-links" activeClassName="underline" to={route}>
+                {page}
+              </NavLink>
+            ))}
+          </li>
```
Comment (Cubified) [inline frontend/src/components/NavBar.js:40]:
```suggestion
          </div>
```

File: frontend/src/components/NavBar.js [hunk 7]
```diff
@@ -1,17 +1,92 @@
...
+        <li className="logo-and-pages">
+          <li className="logo-container">
+            <NavLink to={SITE_PAGES.MANAGE}>
+              <img src="/img/maw_logo.png" alt="Make-A-Wish logo" className="maw-logo" />
+            </NavLink>
+          </li>
+
+          <li className="pages-container">
+            {Object.entries(PAGES).map(([page, route]) => (
+              <NavLink className="page-links" activeClassName="underline" to={route}>
```
Comment (Cubified) [inline frontend/src/components/NavBar.js:36]: React also gives an error if you don't give every element inside `map` a unique key prop (also helps with performance)

```suggestion
              <NavLink key={route} className="page-links" activeClassName="underline" to={route}>
```

File: frontend/src/components/Search.js [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+              alt="Close Search Results"
+              className="close-icon"
+            />
+          </button>
+
+          {input === "" ? (
+            <div>
+              <p className="files-title">All files</p>
+              {allFiles.map((val, index) => (
+                <FileResult val={val} index={index} />
```
Comment (Cubified) [inline frontend/src/components/Search.js:94]: Same thing as in `NavBar.js`, React throws an error if you don't provide a key prop.

```suggestion
                <FileResult key={val.id} val={val} index={index} />
```

File: frontend/src/components/Search.js [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+              </p>
+              <p style={{ fontSize: "12px", margin: 0 }}>Please enter a new search keyword.</p>
+            </div>
+          ) : (
+            <div>
+              <p className="files-title">
+                All files with keyword <q>{input}</q>
+              </p>
+              {filteredFiles.map((val, index) => (
+                <FileResult val={val} index={index} />
```
Comment (Cubified) [inline frontend/src/components/Search.js:111]:
```suggestion
                <FileResult key={val.id} val={val} index={index} />
```

File: frontend/src/styles/Search.css [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+
+.search-input {
+  height: 21px;
+  padding-bottom: 5px;
+  padding-top: 4px;
+  padding-left: 4.95px;
+  font-size: 14px;
+  border: 0px;
+  border-radius: 3px 0px 0px 3px;
+  z-index: 101;
```
Comment (Cubified) [inline frontend/src/styles/Search.css:18]: Input fields don't inherit the font from their parent, so you have to specify it for the input itself.

```suggestion
  z-index: 101;
  font-family: 'Lato', sans-serif;
```

File: frontend/src/styles/Search.css [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+  max-height: 630px;
+  background-color: var(--grey-white);
+  box-shadow: 0px 0px 10px rgba(0, 0, 0, 0.25);
+  overflow: scroll;
+  padding-right: 30px;
+  padding-left: 30px;
+  padding-top: 10px;
+  padding-bottom: 10px;
+  margin-top: 73px;
+  z-index: 100;
```
Comment (Cubified) [inline frontend/src/styles/Search.css:49]:
```suggestion
  z-index: 100;
  font-family: 'Lato', sans-serif;
```

File: frontend/src/components/Search.js (hunk 1 above)
Comment (Cubified) [reply frontend/src/components/Search.js:21]: I tried it, you can pretty much drop the `<FileEntry>` tag from FileEntry.js in place of the `<FileResult>` tag and have it work -- the only thing that needs changing (other than renaming the props to match) is the width of the modal content.

**Review by Cubified — COMMENTED** [review]
Looks good, mostly just some React errors/warnings I left notes about -- you should be able to commit the suggestions I made right from Github if you'd like.

**Review by Cubified — APPROVED** [review]
LGTM!

**Review by PatrickBrown1 — APPROVED** [review]
Nice work!

**Pull request merged**